File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
  people.sort((a, b) => b - a);
  console.log(people);
  let first = 0;
  let last = people.length - 1;
  let count = 0;
  while (first < last) {
    if (people[first] + people[last] <= limit) {
      first += 1;
      last -= 1;
    } else {
      first += 1;
    }
    count++;
    if (first === last) count++;
  }
  return count;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort(reverse=True)
    count=0
    first=0
    last=len(people)-1
    
    while(first<last):
        if(people[first]+people[last]<=limit):
            first+=1
            last-=1
        else:
            first+=1
            
        count+=1
        
        if(first==last):
            count+=1
        
    return count